tools/asg-hevc/src/block_structures.cpp

```diff
@@ -133,7 +133,7 @@ void QuadTree::GetQuadTreeRefSampleAvailVectorRecur(QuadTreeNode & node, BaseBlo
 
 PatchBlock::PatchBlock(const PatchBlock & rhs) : PatchBlock(static_cast<BaseBlock>(rhs))
 {
-    memcpy(PatchData.get(), rhs.PatchData.get(), sizeof(mfxU8) * m_BWidth * m_BHeight * 3 / 2);
+    std::copy(rhs.PatchData.get(), rhs.PatchData.get() + sizeof(mfxU8) * m_BWidth * m_BHeight * 3 / 2, PatchData.get());
```

Comment:
make ```PatchData``` a ```std::vector``` that will solve the problem natively. Current approach looks too artificial